src/nvim/lua/vim.lua

```diff
@@ -386,4 +386,64 @@ do
   vim.wo = new_win_opt_accessor(nil)
 end
 
+--- Get table of lines with start, end columns for given marks
+---
+-- @param mark1 mark of beginning of range
+-- @param mark2 mark of end of range
+-- @param regtype type of selection that is yanked (:help setreg)
+-- @param boolean indicating whether the selection is end-inclusive
+function vim.marks_to_region(mark1, mark2, regtype, inclusive)
+    local pos1 = vim.fn.getpos(mark1)
+    local buf1, lin1, col1, off1 = pos1[1], pos1[2] - 1, pos1[3] - 1, pos1[4]
+    local pos2 = vim.fn.getpos(mark2)
+    local buf2, lin2, col2, off2 = pos2[1], pos2[2] - 1, pos2[3] - (inclusive and 0 or 1), pos2[4]
+
+    -- in case of block selection, columns need to be adjusted for multibyte characters
+    local bufline
+    if regtype:byte() == 22 then
```

Comment:
local ctrl_v = 22

Note that you can do `regtype == '\22' instead of getting the byte.

Reply:
Note that the regtype is longer in this case (`^V6`, for example, if the block is 6 columns wide), so you do need to match specifically on the first byte.